src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -227,13 +227,33 @@ public RegexFindOptimizations(RegexNode root, RegexOptions options)
 
         /// <summary>When in fixed distance set mode, gets the set and how far it is from the start of the pattern.</summary>
         /// <remarks>The case-insensitivity of the 0th entry will always match the mode selected, but subsequent entries may not.</remarks>
-        public List<(char[]? Chars, string Set, int Distance)>? FixedDistanceSets { get; }
+        public List<FixedDistanceSet>? FixedDistanceSets { get; }
+
+        /// <summary>Data about a character class at a fixed offset from the start of any match to a pattern.</summary>
+        public struct FixedDistanceSet
```

Comment:
NIT: Should this be readonly?

Reply:
The way things are currently structured, fields are set post construction, e.g. https://github.com/dotnet/runtime/pull/76859/files#diff-3a7a4260e3a2cd96d2beb16ed4ec8ac508678b139c67f70a37f81be6433cf610R227.

Reply:
Or alternatively I guess it could also become a `readonly record struct` ?

Reply:
> The way things are currently structured, fields are set post construction

Oh I missed that.